lib/views/quiz/quiz_view.dart

```diff
@@ -0,0 +1,170 @@
+  int questionIndex = 0;
+  int score = 0;
+
+  @override
+  void initState() {
+    super.initState();
+    loadQuestions();
+  }
+
+  Future<void> loadQuestions() async {
+    widget.quiz.questions = await QuizQuestionBean.of(context)
+        .preloadAllRelationships(widget.quiz.questions);
+    setState(() => questionsLoaded = true);
+  }
+
+  Future<void> updateHighScore(int score) async {
+    Quiz quiz = widget.quiz;
+    quiz.highScore = score;
+    await QuizBean.of(context).update(quiz);
+  }
+
+  void handleAnswer(int answerId) {
+    if (currentQuestion.trueFalseQuestion != null) {
+      isCorrect = answerId == (currentQuestion.trueFalseQuestion ? 1 : 0);
+      guess = answerId == 0 ? "False" : "True";
+      answer = currentQuestion.trueFalseQuestion ? "True" : "False";
+    } else {
+      isCorrect = answerId == currentQuestion.correctAnswerId;
+      guess = currentQuestion.answers.firstWhere((a) => a.id == answerId).text;
+      answer = currentQuestion.answers
+          .firstWhere((a) => a.id == currentQuestion.correctAnswerId)
+          .text;
+    }
+
+    if (isCorrect) score++;
+
+    setState(() => showOverlay = true);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: questionIndex < widget.quiz.questions.length
+          ? AppBar(
+              brightness: Brightness.dark,
+              title: Heading(widget.quiz.title),
+              centerTitle: true,
+              leading: Container(),
+              actions: [
+                Padding(
+                  padding: EdgeInsets.all(12.0),
+                  child: Heading(
+                    '${questionIndex + 1}/${widget.quiz.questions.length}',
+                  ),
+                )
+              ],
+              backgroundColor: Theme.of(context).primaryColor,
+            )
+          : null,
+      body: questionsLoaded
+          ? Stack(
+              fit: StackFit.expand,
+              children: [
+                buildQuestionView(),
+                if (showOverlay) buildOverlay(),
+              ],
+            )
+          : Center(child: CircularProgressIndicator()),
+    );
+  }
+
+  Widget buildQuestionView() {
+    if (questionIndex == widget.quiz.questions.length) {
+      bool isHighScore = false;
+      if (score > widget.quiz.highScore) {
+        updateHighScore(score);
+        isHighScore = true;
+      }
+
+      return QuizResult(
+        name: widget.quiz.title,
+        score: score,
+        outOf: widget.quiz.questions.length,
+        isHighscore: isHighScore,
+      );
+    }
+
+    // if the question has an image
+    if (currentQuestion.image != null) {
+      return TextQuestion(
+          question: currentQuestion, answers: buildAnswerTiles());
+    }
+
+    // if any answers have an image
+    if (currentQuestion.answers.any((a) => a.image != null)) {
+      return ImageQuestion(
+          question: currentQuestion, answers: buildAnswerTiles());
+    }
+
```

Comment:
are we going to make it so that the trust cannot mix answer types so it can't be a mix of image, image with captions and text?

Reply:
Probably in the CMS side of things